app/models/manageiq/providers/hawkular/common/event_catcher/stream_mixin.rb

```diff
@@ -17,14 +17,21 @@ def each_batch
 
   def fetch
     new_alerts = []
-    alert_tenants.each do |tenant|
-      log_handle.debug "Fetching tenant [#{tenant}] events using criteria [#{hawkular_alert_criteria}]"
-      new_alerts.concat(@ems.alerts_client(:tenant => tenant).list_alerts(hawkular_alert_criteria))
-    end
+    new_alerts.concat(fetch_criteria(open_alert_criteria))
+    new_alerts.concat(fetch_criteria(resolved_alert_criteria)) if method(:resolved_alert_criteria)
     post_fetch(new_alerts)
     new_alerts
   rescue => err
     log_handle.error "Error capturing alerts #{err}"
     []
   end
+
+  def fetch_criteria(criteria)
+    new_alerts = []
+    alert_tenants.each do |tenant|
```

Comment:
can we do here ?
`alert_tenants.map{ |tenant| @ems.alerts_client(:tenant => tenant).list_alerts(criteria) }.flatten`